### PR TITLE
feat(FileUpload): Added FileUpload component

### LINKED ISSUE #49: Create image upload component
Skjermbilde 2020-09-20 kl  15 06 11

We should create this as a kit component, right?

*NOTICE*: We are tracking this on tabetalt/admin#67 as well. When creating a PR for this, please add `fixes tabetalt/admin#67` too.

### PR DESCRIPTION
**Pull request by kvokov**
image

closes #49

### CODE REVIEW

File: src/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { Button } from 'theme-ui';
+import { Button } from './Button';
```
Comment (cobraz) [inline src/components/Button/Button.stories.tsx:2]: Why this?

File: src/components/FileUpload/FileUpload.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    id,
+    accept,
+    multiple,
+    buttonComponent,
+    validate,
+    upload,
+  } = props;
+
+  const [loading, setLoading] = useState(false); // eslint-disable-line @typescript-eslint/no-unused-vars
+  const [errors, setErrors] = useState([] as string[]); // eslint-disable-line @typescript-eslint/no-unused-vars
```
Comment (cobraz) [inline src/components/FileUpload/FileUpload.tsx:42]: Why are these not used?

File: src/components/FileUpload/FileUpload.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  const onInputChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { files: filesToUpload } = event.target;
+    if (!filesToUpload?.length) {
+      return;
+    }
+    if (validate) {
+      const validationErrors = Array.from(filesToUpload).reduce((acc, file) => {
+        const err = validate(file);
+        if (err) {
+          // @ts-ignore
```
Comment (cobraz) [inline src/components/FileUpload/FileUpload.tsx:56]: Is there a way that we can not have this?

File: src/components/FileUpload/components/FileUploadPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+const styles = {
+  width: 6,
+  height: 6,
+  borderRadius: '16px',
+  overflow: 'hidden',
+  mr: 3,
+  mt: 4,
+};
+
+// TODO: add remove button
```
Comment (cobraz) [inline src/components/FileUpload/components/FileUploadPreview.tsx:19]: add and remove? :D

**Review by cobraz — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.stories.tsx (hunk 1 above)
Comment (kvokov) [reply src/components/Button/Button.stories.tsx:2]: I think it's better way, bc for now we just reexport Button from `theme-ui` there, but we may add some customization to it, and it'll be applied also everywhere Button is used
make sense?
But actually I need button with loading state there, so I'll change this

File: src/components/FileUpload/FileUpload.tsx (hunk 2 above)
Comment (kvokov) [reply src/components/FileUpload/FileUpload.tsx:42]: yeah, it's in progress

File: src/components/FileUpload/components/FileUploadPreview.tsx (hunk 4 above)
Comment (kvokov) [reply src/components/FileUpload/components/FileUploadPreview.tsx:19]: I mean that would be good to have "remove" icon showed on hover on uploaded image preview to remove it, no?

**Comment by cobraz** [comment]
@TanyaDolgopolova Can you take over this branch?

**Comment by TanyaDolgopolova** [comment]
> @TanyaDolgopolova Can you take over this branch?

Sure

**Comment by cobraz** [comment]
@TanyaDolgopolova Let me know when you want me to review this again :)

**Review by TanyaDolgopolova — APPROVED** [review]

**Review by TanyaDolgopolova — APPROVED** [review]

**Review by cobraz — APPROVED** [review]

**Pull request merged**